src/interpreter/vm.cc

```diff
@@ -563,6 +564,80 @@ namespace verona::interpreter
     fatal("Reached unreachable opcode");
   }
 
+  Value VM::opcode_pointer_allocate(uint64_t size)
+  {
+    FieldValue* ptr =
+      static_cast<FieldValue*>(alloc_->alloc(size * sizeof(FieldValue)));
+    for (size_t i = 0; i < size; i++)
+    {
+      new (ptr + i) FieldValue();
+    }
+    return Value::pointer(ptr);
+  }
+
+  void VM::opcode_pointer_free(const Value& ptr, uint64_t size)
+  {
+    check_type(ptr, Value::POINTER);
+
+    for (size_t i = 0; i < size; i++)
+    {
+      check(
+        ptr->pointer[i].is_initialized(),
+        "Cannot free pointer with initialized data");
+    }
+
+    alloc_->dealloc(ptr->pointer, size * sizeof(FieldValue));
+  }
+
+  Value
+  VM::opcode_pointer_get(const Value& parent, const Value& ptr, uint64_t index)
+  {
+    check_type(ptr, Value::POINTER);
+    check_type(parent, {Value::ISO, Value::MUT, Value::IMM});
+    return ptr->pointer[index].read(parent.tag);
+  }
+
+  void VM::opcode_pointer_set(
+    const Value& parent, const Value& ptr, uint64_t index, Value value)
+  {
+    check_type(ptr, Value::POINTER);
+    check_type(parent, {Value::ISO, Value::MUT});
+    VMObject* region = parent->object->region();
+
+    check(
+      ptr->pointer[index].is_initialized(),
+      "Cannot overwrite initialized element");
+    ptr->pointer[index].exchange(alloc_, region, std::move(value));
+  }
+
+  Value VM::opcode_pointer_swap(
+    const Value& parent, const Value& ptr, uint64_t index, Value value)
+  {
+    check_type(ptr, Value::POINTER);
+    check_type(parent, {Value::ISO, Value::MUT});
+    VMObject* region = parent->object->region();
+    Value old = ptr->pointer[index].exchange(alloc_, region, std::move(value));
+    return std::move(old);
+  }
+
+  void VM::opcode_pointer_move(
```

Comment:
Do we need to expose this, versus having a loop in Verona code using `swap` and `set`

Reply:
We would need an additional method on Pointer, which extracts a value and leaves the value unitialized. On other hand this method can also be used to implement swap, so it's probably worth doing it.